lib/foreman_remote_execution_core/script_runner.rb

```diff
@@ -178,10 +172,8 @@ def prepare_start
     def initialization_script
       # pipe the output to tee while capturing the exit code in a file
       <<-SCRIPT.gsub(/^\s+\| /, '')
-      | sh <<WRAPPER
-      | (#{@user_method.cli_command_prefix}#{@remote_script} < /dev/null; echo \\$?>#{@exit_code_path}) | /usr/bin/tee #{@output_path}
-      | exit \\$(cat #{@exit_code_path})
-      | WRAPPER
+      | sh -c "(#{@user_method.cli_command_prefix}#{@user_method.instance_of?(ForemanRemoteExecutionCore::SuUserMethod) ? "'#{@remote_script} < /dev/null '" : "#{@remote_script} < /dev/null"}; echo \\$?>#{@exit_code_path}) | /usr/bin/tee #{@output_path}
```

Comment:
Could we pull the `@user_method.instance_of? ...` conditional out of the `SCRIPT`, it is getting a bit hard to read

Reply:
Ok. I will do it.